### ISSUE TITLE
choose a more recent framework to replace Backbone.Marionette+Handlebars

### ISSUE THREAD

**Issue by maxlath** [inventaire/inventaire-client#201]
I don't think this is exactly urgent, but 

Good reasons to stop waiting:
* we might be missing out on some serious workflow benefits [to be listed]
* the more we wait, the more code will need to be converted

Good reasons to wait more:
* the more we wait, the less we will be affected by breaking changes going on the meantime (ex: VueJS v3): how many generations of framework can be spare to learn, by just keep doing what we know :D
* #200 needs to be done first (doing both transitions at once doesn't seem like a good idea)

Inspiration to go through this process:
* [RFC: Adopt a modern JavaScript framework for use with MediaWiki](https://phabricator.wikimedia.org/T241180) (summary: [Wikimedia developers dismiss React for JavaScript makeover](https://www.theregister.co.uk/2020/03/20/wikimedia_react_javascript/))

**Comment by maxlath** [comment]
VueJS growingly seems like the right option:
* getting mature+strong community
* Wikimedia community committed to use it
* part of our team is already familiar with it

proposed steps:
* replace brunch with parcel + convert coffeescript to ES6+ #200 
* once [VueJS v3](https://madewithvuejs.com/blog/vue-3-roundup) is out, start playing with it, possibly producing some layouts with it within the client: the progressive integration is one of its advertised features, and that would give the opportunity to change our mind in the process, without having to pay the cost of a full rewrite. This experimentation should, in theory, be quite cheap, thank to VueJS+parcel out-of-the-box integration

**Comment by maxlath** [comment]
or maybe [Svelte](https://svelte.dev/), here is some good propaganda: [why svelte](https://github.com/feltcoop/why-svelte)

**Comment by maxlath** [comment]
[React, Vue, and Svelte Templates Side By Side](https://medium.com/better-programming/react-vue-and-svelte-templates-side-by-side-4aa52cf3cf2)

**Comment by maxlath** [comment]
So Svelte it is! New layouts are now systematically written in Svelte, and some Backbone.Marionette+Handlebars layouts are being rewritten in Svelte too. Once all Backbone.Marionette+Handlebars layouts have been rewritten, the question of the app router and event bus will remain, but that's a matter for another issue
